autogluon/utils/tabular/features/generators/abstract.py

```diff
@@ -461,11 +461,21 @@ def _remove_features_in(self, features: list):
             List of feature names to remove from the expected input.
         """
         if features:
+            if self._feature_metadata_before_post:
+                feature_links_chain = self.get_feature_links_chain()
+                for feature in features:
+                    feature_links_chain[0].pop(feature)
+                features_to_keep = set()
+                for features_out in feature_links_chain[0].values():
+                    features_to_keep = features_to_keep.union(features_out)
+                self._feature_metadata_before_post = self._feature_metadata_before_post.keep_features(features_to_keep)
+
             self.feature_metadata_in = self.feature_metadata_in.remove_features(features=features)
             self.features_in = self.feature_metadata_in.get_features()
+            if self._pre_astype_generator:
+                self._pre_astype_generator.remove_features_out(features)
 
-    # TODO: Use this to prune features post-fit
-    def _remove_features_out(self, features: list):
+    def remove_features_out(self, features: list):
         """
         Removes features from all relevant objects which represent the content of the output data.
```

Comment:
maybe clarify: "This is used for XYZ".

Reply:
added clarification and made private again since this is advanced functionality that shouldn't be used by many users.